pkg/gitproviders/factory.go

```diff
@@ -30,13 +31,11 @@ type Config struct {
 	Token string
 }
 
-func buildGitProvider(config Config) (gitprovider.Client, error) {
+func buildGitProvider(config Config) (gitprovider.Client, string, error) {
```

Comment:
```suggestion
func buildGitProvider(config Config) (gitprovider.Client, url.URL, error) {
```
^^ Possible? A tad more expressive.

Reply:
It's possible, but the receiving functions take a `string`.  I was trying to make the change as low friction as possible, it did make me sad to add an arbitrary return parameter on that function.